workflow/executor/agent.go

```diff
@@ -173,36 +195,36 @@ func (ae *AgentExecutor) patchWorker(ctx context.Context, taskSetInterface v1alp
 			// Patch was successful, clear nodeResults for next iteration
 			nodeResults = map[string]wfv1.NodeResult{}
 
-			log.WithField("taskset", obj).Infof("Patched TaskSet")
+			log.Info("Patched TaskSet")
 		}
 	}
 }
 
-func (ae *AgentExecutor) processTask(ctx context.Context, tmpl wfv1.Template) (*wfv1.NodeResult, error) {
+func (ae *AgentExecutor) processTask(ctx context.Context, tmpl wfv1.Template) (*wfv1.NodeResult, time.Duration, error) {
+	var executeTemplate templateExecutor
```

Comment:
FYI: I make a lot of similar improvements to this file in https://github.com/argoproj/argo-workflows/pull/7303, so you'll probably face some conflicts